grype/db/v6/store.go

```diff
@@ -0,0 +1,61 @@
+package v6
+
+import (
+	"fmt"
+
+	"gorm.io/gorm"
+
+	"github.com/anchore/grype/grype/db/internal/gormadapter"
+	"github.com/anchore/grype/internal/log"
+)
+
+type store struct {
+	*dbMetadataStore
+	db     *gorm.DB
+	config Config
+	write  bool
+}
+
+func newStore(cfg Config, write bool) (*store, error) {
+	db, err := gormadapter.Open(cfg.DBFilePath(), write)
+	if err != nil {
+		return nil, err
+	}
+
+	db.Exec("PRAGMA foreign_keys = ON")
```

Comment:
notes from sync review:

Why isn't this in the list of PRAGMA statements at `grype/db/internal/gormadapter/open.go` line 15 or so?

Answer: This `PRAGMA` only applies to v6; the others apply to all schemas. It's probably safe to move though - the other schemas lack foreign keys, so the pragma should have no effect.

Reply:
I missed this during the review: those pragmas are for writers only, but we want FKs for readers and writers

Reply:
but the spirit of the comment is essentially: move these pragmas within the `Open` call -- I'll make that change